File: agentic-adoption-scan/output.go
package main

import (
	"encoding/csv"
	"fmt"
	"io"
)

// WriteScanCSV writes scan results in tidy data format.
func WriteScanCSV(w io.Writer, results []ScanResult) error {
	writer := csv.NewWriter(w)
	defer writer.Flush()

	header := []string{
		"scan_timestamp",
		"org",
		"repo",
		"repo_visibility",
		"repo_language",
		"repo_pushed_at",
		"category",
		"indicator",
		"found",
		"file_path",
		"details",
	}

	if err := writer.Write(header); err != nil {
		return fmt.Errorf("writing CSV header: %w", err)
	}

	for _, r := range results {
		found := "false"
		if r.Found {
			found = "true"
		}

		row := []string{
			r.ScanTimestamp,
			r.Org,
			r.Repo,
			r.RepoVisibility,
			r.RepoLanguage,
			r.RepoPushedAt,
			r.Category,
			r.Indicator,
			found,
			r.FilePath,
			r.Details,
		}

		if err := writer.Write(row); err != nil {
			return fmt.Errorf("writing CSV row: %w", err)
		}
	}

	return nil
}

// WriteInspectCSV writes inspection results as CSV.
func WriteInspectCSV(w io.Writer, results []InspectResult) error {
	writer := csv.NewWriter(w)
	defer writer.Flush()

	header := []string{
		"scan_timestamp",
		"org",
		"repo",
		"category",
		"indicator",
		"file_path",
		"content_size",
		"content_summary",
		"raw_content",
	}

	if err := writer.Write(header); err != nil {
		return fmt.Errorf("writing CSV header: %w", err)
	}

	for _, r := range results {
		row := []string{
			r.ScanTimestamp,
			r.Org,
			r.Repo,
			r.Category,
			r.Indicator,
			r.FilePath,
			fmt.Sprintf("%d", r.ContentSize),
			r.ContentSummary,
			r.RawContent,
		}

		if err := writer.Write(row); err != nil {
			return fmt.Errorf("writing CSV row: %w", err)
		}
	}

	return nil
}

File: agentic_adoption_scan/output.py
from __future__ import annotations

import csv
from typing import IO

from .models import InspectResult, ScanResult


def write_scan_csv(writer: IO[str], results: list[ScanResult]) -> None:
    """Write scan results in tidy data format (11-column CSV)."""
    w = csv.writer(writer)

    header = [
        "scan_timestamp",
        "org",
        "repo",
        "repo_visibility",
        "repo_language",
        "repo_pushed_at",
        "category",
        "indicator",
        "found",
        "file_path",
        "details",
    ]
    w.writerow(header)

    for r in results:
        w.writerow([
            r.scan_timestamp,
            r.org,
            r.repo,
            r.repo_visibility,
            r.repo_language,
            r.repo_pushed_at,
            r.category,
            r.indicator,
            "true" if r.found else "false",
            r.file_path,
            r.details,
        ])


def write_inspect_csv(writer: IO[str], results: list[InspectResult]) -> None:
    """Write inspection results as CSV (9-column CSV)."""
    w = csv.writer(writer)

    header = [
        "scan_timestamp",
        "org",
        "repo",
        "category",
        "indicator",
        "file_path",
        "content_size",
        "content_summary",
        "raw_content",
    ]
    w.writerow(header)

    for r in results:
        w.writerow([
            r.scan_timestamp,
            r.org,
            r.repo,
            r.category,
            r.indicator,
            r.file_path,
            str(r.content_size),
            r.content_summary,
            r.raw_content,
        ])
